src/combineReducers.js

```diff
@@ -134,6 +134,7 @@ export default function combineReducers(reducers) {
       nextState[key] = nextStateForKey
       hasChanged = hasChanged || nextStateForKey !== previousStateForKey
     }
+    hasChanged = hasChanged || Object.keys(state).length !== finalReducerKeys.length
```

Comment:
@colinwahl if you have some💡 feel free to spin up a PR. I'll update this PR to fix the development case in the meantime.